### DISCUSSION TITLE
RPM for non-supported subarch

### DISCUSSION THREAD

**Discussion by loximann** [rpm-software-management/rpm#3501] [Q&A]
I need to publish packages built with `-march=znver2` (along with generic x86_64), and I am wondering how to go about this. As far as I have understood digging from [PR 1035](https://github.com/rpm-software-management/rpm/pull/1035), setting BuildArch is not the way to address this.
The only think I can think of is building using `-march=znver2` and overriding `%_rpmfilename`.
Any thoughts, suggestions, caveats?

**Comment by pmatilai** [comment]
There are no good ways to package non-supported arches.

Why do you need znver2 specifically to begin with?

**Reply by loximann** [reply]
Well, we need to support very specific systems, and it would be good to build specifically for those, to squeeze some more performance. It's quite possible it's some Ryzen. We could look into targeting an equivalent arch, but of course picking znverX is the simplest.

**Reply by pmatilai** [reply]
Except that znver is not the simplest because rpm doesn't support it, you'll easily end with pretty bad user experience if you can install a package on an arch that can't actually run it.

If you're targeting rpm >= 4.19 based systems, I'd pick the best fully compatible x86_64-vX subarch (see https://github.com/rpm-software-management/rpm/pull/2315) as package architecture and then maybe build with -mtune=znver2 (instead of -march). Assuming that makes any measurable difference someplace, that is. 

For older rpm's there simply aren't any good options.

Edit: that's all assuming a general purpose OS where users can download and install packages manually. If it's more of an image thing where you control it all then that's different. Depends on the exact use-case scenario.

**Reply by loximann** [reply]
That's correct, we control it all. If anything, we can tolerate that someone gets in trouble because of adding a repo they shouldn't have.
And yes, getting extra performance that is worth all the extra hassle is something that remains to be seen, but my first attempt to start testing that was to just replace the arch in our RPM spec. I'll do the testing by hand then, and we'll see about publishing to a repo later.
Thanks a lot!

**Reply by pmatilai** [reply]
Right, in that case you I suppose the easiest thing is to just add znver2 to your local rpm config as an x86_64 compatible arch with its own build flags. That way you'll at least get the build side as per rpm's expectation. Since the compatibility is handled elsewhere, don't need to come up with asm fu to actually detect the arch, it's enough if *your* rpm has it listed as x86_64 compatible.
